src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/DotNetSdkHelper.cs

```diff
@@ -180,7 +180,14 @@ void processConfigCallback(Process process)
             {
                 if (e.Data?.Contains("Application started. Press Ctrl+C to shut down.") ?? false)
                 {
+#if Windows
+                    [DllImport("Kernel32.dll")]
```

Comment:
I think you can do this without the OS specific code. What about Process.Kill?

`GetProcessByID` https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.getprocessbyid?view=net-7.0 then `Process.Kill` https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=net-7.0

Reply:
The issue I found is that Process.Kill() counts as an abnormal exit and throws a bad exit code. This causes ValidatedExitCode() in ExecuteHelper.cs to fail, which isn't something we want to skip over from my understanding as it might mean things are crashing and we wouldn't notice.

Reply:
Could we propagate to the validation that we are killing the process on purpose somehow?

Reply:
Ok so my current idea is this: The existing code does let me set the expected error as part of ExecuteCmd(), which is in ExecuteRunWeb(). I'll set that to -1 which is produced by Process.Kill(). 

The Web Code should hit it's trigger that it wants to close: (e.Data?.Contains("Application started. Press Ctrl+C to shut down.") ?? false) 
If it doesn't, it should hit the 30000 millisecond timeout which presumably is a different error. 

Arguably this might be enough here. I think it will either crash before the trigger and hit the timeout, or reaches the trigger and Process.Kill() goes through.

We could add one more final check to make sure the Process is still running before Process.Kill() is run and throw an exception if it doesn't, which would be separate from the ValidateExitCode.

Reply:
Either the timeout code doesn't do anything if it is reached, or it's not being caught somehow.

Reply:
Timeout code is also already using Process.Kill() in a different spot so still need to change one of the 2 methods, figure out a way to change the error code, or find a better way to propagate.

Reply:
Based one latest batch of failures, it looks like the exit code for Linux/Mac is not -1 like it is on Windows.  All web tests for those os' failed with this exception. Meaning we either need to have separate exit code checks based on each OS or we really need do need to skip it.
System.InvalidOperationException : Failed to execute /Users/runner/work/1/s/.dotnet/dotnet run /bl:/var/folders/3s/vfzpb5r51gs6y328rmlgzm7c0000gn/T/ZhDEa8/SdkTemplateTest_Mvc_CSharp/run.binlog
  Exit code: 137